slack-plugin/src/main/kotlin/slack/ProcessExtensions.kt

```diff
@@ -16,51 +16,123 @@
 package slack
 
 import java.io.File
+import javax.inject.Inject
+import okio.Buffer
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.ListProperty
 import org.gradle.api.provider.Provider
 import org.gradle.api.provider.ProviderFactory
+import org.gradle.api.provider.ValueSource
+import org.gradle.api.provider.ValueSourceParameters
+import org.gradle.api.tasks.Optional
+import org.gradle.process.ExecOperations
+import org.gradle.process.ExecSpec
+import org.gradle.process.internal.ExecException
+import slack.ExecSource.Parameters
 import slack.gradle.util.sneakyNull
 
-internal fun String.executeBlocking(providers: ProviderFactory, workingDir: File) {
-  executeBlockingWithResult(providers, workingDir)
+internal fun String.executeBlocking(
+  providers: ProviderFactory,
+  workingDir: File,
+  isRelevantToConfigurationCache: Boolean
```

Comment:
Most of this should "just work" :) :
- if there are properties read during configuration, then if their value changes, you probably want to re-run, right?
- if there are properties that you need to read, but they aren't really impacting conf (such as build scan tags), that sounds like a missing API

If this provides huge decrease in configuration cache misses, it could be worth merging, but I think it may cause quite a lot of hard-to-debug bugs.

Reply:
That is a very interesting and sad find... :/

Yeah, keeping this as-is for now seems reasonable, but I'd consider filing an issue on your end to keep track of progress on https://github.com/gradle/gradle-org-conventions-plugin/issues/18, and remove this once that lands.